File: .agents/skills/create-agent-python/templates/agent_name.py
from shared.<domain>.taxonomy_<name>_vo import <VO>
from shared.<domain>.contract_<name>_aggregate import I<Name>Aggregate
from shared.<domain>.contract_<protocol>_protocol import I<Protocol>Protocol


# ─── Block 1: Struct Definition & Constructor ──────────────
class <Name>Orchestrator(I<Name>Aggregate):
    def __init__(self, service: I<Protocol>Protocol) -> None:
        # DI fields use protocol interfaces
        # Value fields use shared VOs
        self._service = service

    # ─── Block 2: Public Contract (domain aggregate ONLY) ──
    def execute(self, request: <RequestVO>) -> <ResultVO>:
        # orchestration only - delegate to protocol
        return self._service.process(request)

    # ─── Block 3: Dunder Methods, Factories & Helpers ─────
    def __repr__(self) -> str:
        return "<Name>Orchestrator()"

    @classmethod
    def create_default(cls) -> "<Name>Orchestrator":
        return cls(service=Default<Protocol>())

File: .agents/skills/create-agent-rust/templates/agent_name.rs
use std::sync::Arc;

use shared::<domain>::taxonomy_<name>_vo::<VO>;
use shared::<domain>::contract_<name>_aggregate::I<Name>Aggregate;
use shared::<domain>::contract_<protocol>_protocol::I<Protocol>Protocol;

// ─── Block 1: Struct Definition ───────────────────────────
pub struct <Name>Orchestrator {
    // DI fields use Arc<dyn Trait>
    // Value fields use shared VOs
    service: Arc<dyn I<Protocol>Protocol>,
}

// ─── Block 2: Aggregate Trait Implementation ──────────────
impl I<Name>Aggregate for <Name>Orchestrator {
    fn execute(&self, request: &<RequestVO>) -> <ResultVO> {
        // orchestration only - delegate to protocol
        self.service.process(request)
    }
}

// ─── Block 3: Constructors, Helpers, Private Methods ──────
impl <Name>Orchestrator {
    pub fn new(service: Arc<dyn I<Protocol>Protocol>) -> Self {
        Self { service }
    }
}
